### ISSUE TITLE
Improvements to disabling anonymous access to Devportal

### ISSUE THREAD

**Issue by tmkasun** [wso2/product-apim#9722]
### Describe your problem(s)
When accessing the Developer portal in APIM manager, We don't authenticate the user nor check the authorization (simply don't require any logins) to access the Developer portal. But in some cases, users might need to prevent such anonymous access to the Developer Portal, This should be allowed (disable anonymous access) through a server configuration and it should be enforced in the REST API as well means when the anonymous access is disabled anyone should not be allowed access the Devportal UI as well as REST APIs without proper login or access token.

### Describe your solution

In the current implementation, Anonymous access to the devportal can be disabled through a configuration in 
```
/_system/config/apimgt/applicationdata/tenant-conf.json
```
using `EnableAnonymous` property
```json
{
  "EnableMonetization" : false,
  "EnableAnonymous": false,
  "EnableRecommendation" : false,
.
.
.
}

```
 But that does not prevent REST API access to anonymous users.

for example, users can still get the list of APIs using the following API without any credentials

```
curl 'https://localhost:9443/api/am/store/v1/apis?limit=10&offset=0' \
  -H 'Connection: keep-alive' \
  -H 'Pragma: no-cache' \
  -H 'Cache-Control: no-cache' \
  -H 'sec-ch-ua: "Google Chrome";v="87", " Not;A Brand";v="99", "Chromium";v="87"' \
  -H 'DNT: 1' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36' \
  -H 'Accept: */*' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Referer: https://localhost:9443/devportal/apis' \
  -H 'Accept-Language: en-US,en;q=0.9' \
```

### How will you implement it

- We need to use the above configuration to restrict the access to store REST APIs when it's configured to disable anonymous access.

**Comment by tmkasun** [comment]
@malinthaprasan 

Regarding the configuration property `EnableAnonymous`, 

Shall we rename this as `DisableAnonymousDevportal` and make the default value(`false`) available in the `tenant-config.json` ?

**Comment by malinthaprasan** [comment]
+1 but we cannot change it without a migration in master branch or support both configs?

**Comment by wasuradananjith** [comment]
This has been fixed for APIM 3.0.0 with [1] by backporting the devportal UI anonymous feature to APIM 3.0.0 from APIM 4.x. Furthermore, the above stated REST API validation has been done in the PreAuthenticationInterceptor.java, thus by resolving the customer issue.

But still this has not been resolved in the master. We can apply the same fix for now to master to just sync, or else need to analyze thoroughly for a proper fix. 

[1] https://github.com/wso2-support/carbon-apimgt/pull/2815
